addons/mail/static/src/core/common/message.js

```diff
@@ -448,6 +457,9 @@ export class Message extends Component {
             },
         };
         this.state.isEditing = true;
+        Promise.all(mentionedChannels).then((mentionedChannels) => {
+            this.props.message.composer.mentionedChannels.add(...mentionedChannels);
+        });
```

Comment:
If you edit quickly then save, this won't take into account.
And it could even be added to another composer, and mistakenly notify channel is mentioned (ongoing task).

The save on edit should await this. Abort of editing should stop this behaviour from happening in any following composer (except on composer with same identity, not really a problem in this case)

Reply:
actually, adding to another composer is not a problem because we will check the valid mentions when saving and all the invalid mentions will be discarded. And the quick save can be a problem indeed ( or when the network is extremely slow ). I will fix that, thanks.

Reply:
@zel-odoo Who knows someone uses `#` without an actual channel mention, e.g. for jokingly putting some tags in body message as if this was social network message. Would feel bad that they collide by mistake with an actual channel named that way.

This is very unlikely, but possible :P

Reply:
Ok, so I pushed the new code and all the `Prpmise`s should go with the message directly, rather than with the composer. That's what the recipients do, sadly the mentioned channels are not stored in the database so we have to parse from the user input and getOrfetch it again when needed.